### PR TITLE
[QOL] Candy Upgrade Notification Rework

### PR DESCRIPTION
**Pull request by josericardo-fo**
## What are the changes?

Remake of #2808 
Reworked the Candy Upgrade Notification logic to take account of the new "Buy Egg" option.

## Why am I doing these changes?

A new mechanic was implemented (Buying Eggs).

## What did change?

The logic of the 'On' setting.

### Before (Look at Spearow)
image

### After
image

## How to test the changes?

Test if the candy icon appears whenever you can buy eggs.

## Checklist
- [X] There is no overlap with another PR?
- [X] The PR is self-contained and cannot be split into smaller PRs?
- [X] Have I provided a clear explanation of the changes?
- [X] Have I tested the changes (manually)?
    - [X] Are all unit tests still passing? (`npm run test`)
- [X] Are the changes visual?
  - [X] Have I provided screenshots/videos of the changes?

### CODE REVIEW

**Comment by Fontbane** [comment]
Would it be possible to only show this notification after both cost reductions were bought? It would be mildly annoying imo to be grinding for a cost reduction and see the notification, only to find out that you still don't have enough for the cost reduction, just enough to buy an egg. This is kind of the reasoning behind the "Passives Only" option for notifications.

File: src/ui/starter-select-ui-handler.ts [hunk 1]
```diff
@@ -119,6 +119,8 @@ const starterCandyCosts: { passive: integer, costReduction: [integer, integer],
   { passive: 10, costReduction: [3, 10], egg: 10 },  // 10
 ];
 
+export const VALUE_REDUCTION_MAX = 2;
```
Comment (torranx) [inline src/ui/starter-select-ui-handler.ts:122]: does this need to be an export?

Comment (josericardo-fo) [reply src/ui/starter-select-ui-handler.ts:122]: you're right

**Comment by josericardo-fo** [comment]
> Would it be possible to only show this notification after both cost reductions were bought? It would be mildly annoying imo to be grinding for a cost reduction and see the notification, only to find out that you still don't have enough for the cost reduction, just enough to buy an egg. This is kind of the reasoning behind the "Passives Only" option for notifications.

I don't really know how should I implement that. It would make more sense to add another option, but some languages don't even have enough space to fit it. Any suggestions?

**Comment by josericardo-fo** [comment]
@Fontbane There's a new filter in the starter screen now. No need for these type of filters in the setting anymore.

**Comment by josericardo-fo** [comment]
Even with the addition of the new filters, this should be a thing changed in the settings

**Comment by torranx** [comment]
This PR needs some fixing - it carries past commits from other people

**Comment by josericardo-fo** [comment]
> This PR needs some fixing - it carries past commits from other people

Moved to #3720

**Pull request closed without merging**